web-app/js/portal/cart/DownloadPanel.js

```diff
@@ -177,9 +196,42 @@ Portal.cart.DownloadPanel = Ext.extend(Ext.Panel, {
                     scope: this
                 };
 
-                processedValues.menuItems.push(newMenuItem);
+                if (hasDuplicateWFSDownloads) {
+                    if (this.getEmbededTitle(handler.onlineResource.title)) {
+                        var groupLabel = this.getEmbededTitle(handler.onlineResource.title);
+                        if (groupedMenuItems[groupLabel] == undefined) {
+                            groupedMenuItems[groupLabel] = [];
+                        }
+                        groupedMenuItems[groupLabel].push(newMenuItem);
+                    }
+                    else if (menuItemLabelsUsed.indexOf(newMenuItem.text) == -1) {
+                        processedValues.menuItems.push(newMenuItem);
+                    }
+                }
+                else {
+                    processedValues.menuItems.push(newMenuItem);
+                }
+
+                menuItemLabelsUsed.push(newMenuItem.text);
+
             }, this);
         }, this);
+
+        // grouped items will appear last
+        for (var key in groupedMenuItems) {
+            if (groupedMenuItems.hasOwnProperty(key)) {
+                processedValues.menuItems.push(OpenLayers.i18n('menuItemGroupTitle', {title: key}));
+                Ext.each(groupedMenuItems[key], function(item) {
+                    processedValues.menuItems.push(item);
+                });
+            }
+        };
+    },
+
+    getEmbededTitle: function(title) {
```

Comment:
spelling mistake - "embedded".  Sorry, I know that's pedantic, but it's annoying having to repeatedly "correct" something to the wrong spelling when you're using it. :-)